### ISSUE TITLE
[Select] allow option groups

### ISSUE THREAD

**Issue by crimeminister** [telus/tds-core#1276]
## Description

I have a list of 10-15 items that belong to several categories. Rather than prefix each item with a label like so:

- `Category A: Item 1`
- `Category A: Item 2`
- `Category B: Item 1`
- `Category B: Item 2`

I would prefer to include labels in the list to serve as headers, in order to separate the elements of different categories, e.g.

- `Category A`
    - `Item 1`
    - `Item 2`
- `Category B`
    - `Item 1`
    - `Item 2`

## Recommendation

I would like to be able to mark list items as being labels, e.g.

```
<Select
  label="My Label"
  placeholder="Please select..."
  options={[
    { text: "Category A", label: true },
    { text: "Item 1", value: "item_a1" },
    { text: "Item 2", value: "item_a2" },
    { text: "Category B", label: true },
    { text: "Item 1", value: "item_b1" },
    { text: "Item 2", value: "item_b2" }
  ]}
  defaultValue="item_a1"
  />
```

## Designs

I imagine that items marked as labels above would receive the same visual treatment as the item specified as the value of the `placeholder` attribute, i.e. be turned into a list item that is `disabled`.

## Meta

- TDS component:  `@tds/core-select`
- Willing to develop solution: Yes
- Has workaround: Yes (using label prefixes in list items)
- High impact: No

**Comment by varunj90** [comment]
@crimeminister as discussed on slack, we'd love to see a PR for this.

thanks!

**Comment by theetrain** [comment]
This feature was released in [@tds/core-select@3.2.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-select%403.2.0).

🎉
